// @vercel/sdk/esm/models/deleteconfigurablelogdrainop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const DeleteConfigurableLogDrainRequest$inboundSchema = z.object({
    id: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/** @internal */
export const DeleteConfigurableLogDrainRequest$outboundSchema = z.object({
    id: z.string(),
    teamId: z.string().optional(),
    slug: z.string().optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var DeleteConfigurableLogDrainRequest$;
(function (DeleteConfigurableLogDrainRequest$) {
    /** @deprecated use `DeleteConfigurableLogDrainRequest$inboundSchema` instead. */
    DeleteConfigurableLogDrainRequest$.inboundSchema = DeleteConfigurableLogDrainRequest$inboundSchema;
    /** @deprecated use `DeleteConfigurableLogDrainRequest$outboundSchema` instead. */
    DeleteConfigurableLogDrainRequest$.outboundSchema = DeleteConfigurableLogDrainRequest$outboundSchema;
})(DeleteConfigurableLogDrainRequest$ || (DeleteConfigurableLogDrainRequest$ = {}));
export function deleteConfigurableLogDrainRequestToJSON(deleteConfigurableLogDrainRequest) {
    return JSON.stringify(DeleteConfigurableLogDrainRequest$outboundSchema.parse(deleteConfigurableLogDrainRequest));
}
export function deleteConfigurableLogDrainRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => DeleteConfigurableLogDrainRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DeleteConfigurableLogDrainRequest' from JSON`);
}
