// @vercel/sdk/esm/models/getprojectsprojectidlogspresetsop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
/** @internal */
export const GetProjectsProjectIdLogsPresetsRequest$inboundSchema = z.object({
    projectId: z.string(),
});
/** @internal */
export const GetProjectsProjectIdLogsPresetsRequest$outboundSchema = z.object({
    projectId: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetProjectsProjectIdLogsPresetsRequest$;
(function (GetProjectsProjectIdLogsPresetsRequest$) {
    /** @deprecated use `GetProjectsProjectIdLogsPresetsRequest$inboundSchema` instead. */
    GetProjectsProjectIdLogsPresetsRequest$.inboundSchema = GetProjectsProjectIdLogsPresetsRequest$inboundSchema;
    /** @deprecated use `GetProjectsProjectIdLogsPresetsRequest$outboundSchema` instead. */
    GetProjectsProjectIdLogsPresetsRequest$.outboundSchema = GetProjectsProjectIdLogsPresetsRequest$outboundSchema;
})(GetProjectsProjectIdLogsPresetsRequest$ || (GetProjectsProjectIdLogsPresetsRequest$ = {}));
export function getProjectsProjectIdLogsPresetsRequestToJSON(getProjectsProjectIdLogsPresetsRequest) {
    return JSON.stringify(GetProjectsProjectIdLogsPresetsRequest$outboundSchema.parse(getProjectsProjectIdLogsPresetsRequest));
}
export function getProjectsProjectIdLogsPresetsRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetProjectsProjectIdLogsPresetsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetProjectsProjectIdLogsPresetsRequest' from JSON`);
}
