// @vercel/sdk/esm/models/updateinvoiceop.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../lib/primitives.js";
import { safeParse } from "../lib/schemas.js";
export const UpdateInvoiceAction = {
    Refund: "refund",
};
/** @internal */
export const UpdateInvoiceAction$inboundSchema = z.nativeEnum(UpdateInvoiceAction);
/** @internal */
export const UpdateInvoiceAction$outboundSchema = UpdateInvoiceAction$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateInvoiceAction$;
(function (UpdateInvoiceAction$) {
    /** @deprecated use `UpdateInvoiceAction$inboundSchema` instead. */
    UpdateInvoiceAction$.inboundSchema = UpdateInvoiceAction$inboundSchema;
    /** @deprecated use `UpdateInvoiceAction$outboundSchema` instead. */
    UpdateInvoiceAction$.outboundSchema = UpdateInvoiceAction$outboundSchema;
})(UpdateInvoiceAction$ || (UpdateInvoiceAction$ = {}));
/** @internal */
export const UpdateInvoiceRequestBody$inboundSchema = z.object({
    action: UpdateInvoiceAction$inboundSchema,
    reason: z.string(),
    total: z.string(),
});
/** @internal */
export const UpdateInvoiceRequestBody$outboundSchema = z.object({
    action: UpdateInvoiceAction$outboundSchema,
    reason: z.string(),
    total: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateInvoiceRequestBody$;
(function (UpdateInvoiceRequestBody$) {
    /** @deprecated use `UpdateInvoiceRequestBody$inboundSchema` instead. */
    UpdateInvoiceRequestBody$.inboundSchema = UpdateInvoiceRequestBody$inboundSchema;
    /** @deprecated use `UpdateInvoiceRequestBody$outboundSchema` instead. */
    UpdateInvoiceRequestBody$.outboundSchema = UpdateInvoiceRequestBody$outboundSchema;
})(UpdateInvoiceRequestBody$ || (UpdateInvoiceRequestBody$ = {}));
export function updateInvoiceRequestBodyToJSON(updateInvoiceRequestBody) {
    return JSON.stringify(UpdateInvoiceRequestBody$outboundSchema.parse(updateInvoiceRequestBody));
}
export function updateInvoiceRequestBodyFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateInvoiceRequestBody$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateInvoiceRequestBody' from JSON`);
}
/** @internal */
export const UpdateInvoiceRequest$inboundSchema = z.object({
    integrationConfigurationId: z.string(),
    invoiceId: z.string(),
    RequestBody: z.lazy(() => UpdateInvoiceRequestBody$inboundSchema),
}).transform((v) => {
    return remap$(v, {
        "RequestBody": "requestBody",
    });
});
/** @internal */
export const UpdateInvoiceRequest$outboundSchema = z.object({
    integrationConfigurationId: z.string(),
    invoiceId: z.string(),
    requestBody: z.lazy(() => UpdateInvoiceRequestBody$outboundSchema),
}).transform((v) => {
    return remap$(v, {
        requestBody: "RequestBody",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var UpdateInvoiceRequest$;
(function (UpdateInvoiceRequest$) {
    /** @deprecated use `UpdateInvoiceRequest$inboundSchema` instead. */
    UpdateInvoiceRequest$.inboundSchema = UpdateInvoiceRequest$inboundSchema;
    /** @deprecated use `UpdateInvoiceRequest$outboundSchema` instead. */
    UpdateInvoiceRequest$.outboundSchema = UpdateInvoiceRequest$outboundSchema;
})(UpdateInvoiceRequest$ || (UpdateInvoiceRequest$ = {}));
export function updateInvoiceRequestToJSON(updateInvoiceRequest) {
    return JSON.stringify(UpdateInvoiceRequest$outboundSchema.parse(updateInvoiceRequest));
}
export function updateInvoiceRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => UpdateInvoiceRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'UpdateInvoiceRequest' from JSON`);
}
