// @vercel/sdk/esm/models/vercelratelimiterror.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../lib/schemas.js";
import { RateLimitNotice$inboundSchema, RateLimitNotice$outboundSchema, } from "./ratelimitnotice.js";
export class VercelRateLimitError extends Error {
    constructor(err) {
        const message = "message" in err && typeof err.message === "string"
            ? err.message
            : `API error occurred: ${JSON.stringify(err)}`;
        super(message);
        this.data$ = err;
        this.error = err.error;
        this.name = "VercelRateLimitError";
    }
}
/** @internal */
export const VercelRateLimitErrorError$inboundSchema = z.object({
    code: z.string(),
    message: z.string(),
    limit: RateLimitNotice$inboundSchema.optional(),
});
/** @internal */
export const VercelRateLimitErrorError$outboundSchema = z.object({
    code: z.string(),
    message: z.string(),
    limit: RateLimitNotice$outboundSchema.optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var VercelRateLimitErrorError$;
(function (VercelRateLimitErrorError$) {
    /** @deprecated use `VercelRateLimitErrorError$inboundSchema` instead. */
    VercelRateLimitErrorError$.inboundSchema = VercelRateLimitErrorError$inboundSchema;
    /** @deprecated use `VercelRateLimitErrorError$outboundSchema` instead. */
    VercelRateLimitErrorError$.outboundSchema = VercelRateLimitErrorError$outboundSchema;
})(VercelRateLimitErrorError$ || (VercelRateLimitErrorError$ = {}));
export function vercelRateLimitErrorErrorToJSON(vercelRateLimitErrorError) {
    return JSON.stringify(VercelRateLimitErrorError$outboundSchema.parse(vercelRateLimitErrorError));
}
export function vercelRateLimitErrorErrorFromJSON(jsonString) {
    return safeParse(jsonString, (x) => VercelRateLimitErrorError$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'VercelRateLimitErrorError' from JSON`);
}
/** @internal */
export const VercelRateLimitError$inboundSchema = z.object({
    error: z.lazy(() => VercelRateLimitErrorError$inboundSchema),
})
    .transform((v) => {
    return new VercelRateLimitError(v);
});
/** @internal */
export const VercelRateLimitError$outboundSchema = z.instanceof(VercelRateLimitError)
    .transform(v => v.data$)
    .pipe(z.object({
    error: z.lazy(() => VercelRateLimitErrorError$outboundSchema),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var VercelRateLimitError$;
(function (VercelRateLimitError$) {
    /** @deprecated use `VercelRateLimitError$inboundSchema` instead. */
    VercelRateLimitError$.inboundSchema = VercelRateLimitError$inboundSchema;
    /** @deprecated use `VercelRateLimitError$outboundSchema` instead. */
    VercelRateLimitError$.outboundSchema = VercelRateLimitError$outboundSchema;
})(VercelRateLimitError$ || (VercelRateLimitError$ = {}));
