// @vercel/sdk/esm/mcp-server/tools/deleteProjectsProjectIdLogsPresetsId.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import { deleteProjectsProjectIdLogsPresetsId } from "../../funcs/deleteProjectsProjectIdLogsPresetsId.js";
import { DeleteProjectsProjectIdLogsPresetsIdRequest$inboundSchema } from "../../models/deleteprojectsprojectidlogspresetsidop.js";
import { formatResult } from "../tools.js";
const args = {
    request: DeleteProjectsProjectIdLogsPresetsIdRequest$inboundSchema,
};
export const tool$deleteProjectsProjectIdLogsPresetsId = {
    name: "delete-projects-project-id-logs-presets-id",
    description: ``,
    args,
    tool: async (client, args, ctx) => {
        const [result, apiCall] = await deleteProjectsProjectIdLogsPresetsId(client, args.request, { fetchOptions: { signal: ctx.signal } }).$inspect();
        if (!result.ok) {
            return {
                content: [{ type: "text", text: result.error.message }],
                isError: true,
            };
        }
        return formatResult(void 0, apiCall);
    },
};
